src/main/java/org/spongepowered/common/mixin/api/mcp/entity/projectile/EvokerFangsEntityMixin_API.java

```diff
@@ -0,0 +1,45 @@
+package org.spongepowered.common.mixin.api.mcp.entity.projectile;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.EvokerFangsEntity;
+import org.spongepowered.api.entity.projectile.EvokerFangs;
+import org.spongepowered.api.projectile.source.ProjectileSource;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.mixin.api.mcp.entity.EntityMixin_API;
+
+import javax.annotation.Nullable;
+
+@Mixin(EvokerFangsEntity.class)
+public abstract class EvokerFangsEntityMixin_API extends EntityMixin_API implements EvokerFangs {
+
+    @Shadow private LivingEntity caster;
+
+    // Not all ProjectileSources are entities (e.g. BlockProjectileSource).
+    // This field is used to store a ProjectileSource that isn't an entity.
+    @Nullable
+    public ProjectileSource projectileSource;
```

Comment:
if this is being added, please make sure it's prefixed by the package it's being added from (in this case `api$projectileSource`). It helps when debugging and whatnot to see what fields are being added from where.

Reply:
Keep @Nullable on same line as well.